File: tests/resources/test_codegen/test_default_code_generators_and_templates/cpp/echo_template.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cassert>

{% if mod is not none %}
const int mod = {{ mod }};
{% endif %}
{% if yes_str is not none %}
const std::string YES = "{{ yes_str }}";
{% endif %}
{% if no_str is not none %}
const std::string NO = "{{ no_str }}";
{% endif %}
void solve({{ formal_arguments }}){
    std::cout << N << " " << M << std::endl;
    assert(H.size() == N - 1);
    for (int i = 0; i < N - 1; i++) {
        assert(H[i].size() == M - 2);
        for (int j = 0; j < M - 2; j++) {
            std::cout << (j > 0 ? " " : "") << H[i][j];
        }
        std::cout << std::endl;
    }
    assert(A.size() == N - 1);
    assert(B.size() == N - 1);
    for(int i = 0;i < N - 1;i++){
        std::cout << A[i] << " " << B[i] << std::endl;
    }
    std::cout << Q << std::endl;
    assert(X.size() == M + Q);
    for(int i = 0;i < M + Q;i++){
        std::cout << X[i] << std::endl;
    }

    std::cout << YES << std::endl;
    std::cout << NO << std::endl;
    std::cout << mod << std::endl;

}

int main(){
    {{input_part}}
    solve({{ actual_arguments }});
    return 0;
}

File: tests/resources/test_codegen/test_default_code_generators_and_templates/python/echo_template.py
#!/usr/bin/env python3
{% if prediction_success %}
import sys
{% endif %}
{% if mod or yes_str or no_str %}

{% endif %}
{% if mod %}
MOD = {{ mod }}  # type: int
{% endif %}
{% if yes_str %}
YES = "{{ yes_str }}"  # type: str
{% endif %}
{% if no_str %}
NO = "{{ no_str }}"  # type: str
{% endif %}
{% if prediction_success %}


def solve({{ formal_arguments }}):
    print(N, M)
    assert len(H) == N - 1
    for i in range(N - 1):
        assert len(H[i]) == M - 2
        print(*H[i])
    assert len(A) == N - 1
    assert len(B) == N - 1
    for i in range(N - 1):
        print(A[i], B[i])
    print(Q)
    assert len(X) == M + Q
    for i in range(M + Q):
        print(X[i])

    print(YES)
    print(NO)
    print(MOD)
{% endif %}


def main():
    def iterate_tokens():
        for line in sys.stdin:
            for word in line.split():
                yield word
    tokens = iterate_tokens()
    {{ input_part }}
    solve({{ actual_arguments }})

if __name__ == '__main__':
    main()
